pkg/providers/vsphere/vmprovider.go

```diff
@@ -425,6 +425,32 @@ func (vs *vSphereVMProvider) GetItemFromLibraryByName(ctx context.Context,
 	return contentLibraryProvider.GetLibraryItem(ctx, contentLibrary, itemName, false)
 }
 
+func (vs *vSphereVMProvider) GetItemFromInventoryByName(
+	ctx context.Context,
+	contentLibrary, itemName string) (object.Reference, error) {
+
+	client, err := vs.getVcClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	c := client.VimClient()
+
+	searchIndex := object.NewSearchIndex(c)
+
+	folderRef := vimtypes.ManagedObjectReference{
+		Type:  string(vimtypes.ManagedObjectTypesFolder),
+		Value: contentLibrary,
+	}
+
+	vm, err := searchIndex.FindChild(ctx, folderRef, itemName)
```

Comment:
You may want to look at what error is returned if no child is found and instead of returning an error out of this function, return a `(object.Reference, bool, error)` to indicate there was no error, but the VM wasn't found either. Not sure. @bryanv , thoughts?

Reply:
per https://developer.broadcom.com/xapis/virtual-storage-lifecycle-management-api/9.0/vim.SearchIndex.html?scrollString=FindChild#findChild null is returned if the entity is not found. it does not document any returned faults if the entity is not found.

and from https://github.com/vmware/govmomi/blob/main/object/search_index.go#L154-L161

it looks like either `nil, err`, `nil, nil`, or `object.Reference, nil` is returned from `FindChild`. perhaps i'm missing something obvious, but isn't it enought to know that if we if `nil, nil` is returned, then no entity was found and no fault was returned?